ui/v2.5/src/components/Galleries/GalleryDetails/GalleryChapterForm.tsx

```diff
@@ -0,0 +1,163 @@
+import React from "react";
+import { Button, Form } from "react-bootstrap";
+import { FormattedMessage, useIntl } from "react-intl";
+import { Field, FieldProps, Form as FormikForm, Formik } from "formik";
+import * as GQL from "src/core/generated-graphql";
+import {
+  useGalleryChapterCreate,
+  useGalleryChapterUpdate,
+  useGalleryChapterDestroy,
+} from "src/core/StashService";
+import { useToast } from "src/hooks/Toast";
+
+interface IFormFields {
+  title: string;
+  imageIndex: string;
+}
+
+interface IGalleryChapterForm {
+  galleryID: string;
+  editingChapter?: GQL.GalleryChapterDataFragment;
+  onClose: () => void;
+}
+
+export const GalleryChapterForm: React.FC<IGalleryChapterForm> = ({
+  galleryID,
+  editingChapter,
+  onClose,
+}) => {
+  const [galleryChapterCreate] = useGalleryChapterCreate();
+  const [galleryChapterUpdate] = useGalleryChapterUpdate();
+  const [galleryChapterDestroy] = useGalleryChapterDestroy();
+  const Toast = useToast();
+  const intl = useIntl();
+  const imagesQuery = GQL.useFindImagesQuery({
+    variables: {
+      filter: {
+        per_page: 100000,
+        sort: "path",
+      },
+      image_filter: {
+        galleries: {
+          modifier: GQL.CriterionModifier.Includes,
+          value: [galleryID],
+        },
+      },
+    },
+  });
+
+  const onSubmit = async (values: IFormFields) => {
+    const variables:
+      | GQL.GalleryChapterUpdateInput
+      | GQL.GalleryChapterCreateInput = {
+      title: values.title,
+      image_index: parseInt(values.imageIndex),
+      gallery_id: galleryID,
+    };
+    await imagesQuery;
+    if (
+      imagesQuery.data?.findImages.count &&
+      variables.image_index > imagesQuery.data.findImages.count
+    ) {
+      Toast.error(intl.formatMessage({ id: "toast.image_index_too_large" }));
+      return;
+    }
```

Comment:
This should be performed by the back-end, not the front-end, and it should also validate for negative indexes. We should be able to do formik-based validation on the field as well.

Reply:
Is now done via Backend, a Form has been established to block < 1 and non-numbers.